src/Units/AWBWUnits.java

```diff
@@ -0,0 +1,778 @@
+
+    // Inscribe those war machines obtainable from an Airport.
+    airportModels.add(new TCopterModel());
+    airportModels.add(new BCopterModel());
+    airportModels.add(new FighterModel());
+    airportModels.add(new BomberModel());
+    UnitModel stealth = new StealthModel();
+    airportModels.add(stealth);
+    airportModels.add(new BBombModel());
+
+    // Dump these lists into a hashmap for easy reference later.
+    awbwModels.shoppingList.put(TerrainType.FACTORY, factoryModels);
+    awbwModels.shoppingList.put(TerrainType.SEAPORT, seaportModels);
+    awbwModels.shoppingList.put(TerrainType.AIRPORT, airportModels);
+
+    // Compile one master list of everything we can build.
+    for (UnitModel um : factoryModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : seaportModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : airportModels)
+      awbwModels.unitModels.add(um);
+
+    // Handle transforming units separately, since we don't want two buy-entries
+    UnitModel subsub = new SubSubModel();
+    sub.possibleActions.add(new UnitActionType.Transform(subsub, "DIVE"));
+    subsub.possibleActions.add(new UnitActionType.Transform(sub, "RISE"));
+    awbwModels.unitModels.add(subsub);
+    UnitModel sneaky = new StealthHideModel();
+    stealth.possibleActions.add(new UnitActionType.Transform(sneaky, "HIDE"));
+    sneaky.possibleActions.add(new UnitActionType.Transform(stealth, "APPEAR"));
+    awbwModels.unitModels.add(sneaky);
+
+    return awbwModels;
+  }
+
+  public enum AWBWUnitEnum
+  {
+    INFANTRY, MECH,
+    RECON, TANK, MD_TANK, NEOTANK, MEGATANK,
+    APC, ARTILLERY, ROCKETS, PIPERUNNER, ANTI_AIR, MOBILESAM,
+    FIGHTER, BOMBER, STEALTH, STEALTH_HIDE, B_COPTER, T_COPTER, BBOMB,
+    CARRIER, BBOAT, BATTLESHIP, CRUISER, LANDER, SUB, SUB_SUB
+  };
+
+  public static class AWBWUnitModel extends UnitModel
+  {
+    private static final long serialVersionUID = 1L;
+    public AWBWUnitEnum type;
+
+    public AWBWUnitModel(String pName, AWBWUnitEnum pType, UnitRoleEnum pRole, ChassisEnum pChassis, int cost, int pAmmoMax, int pFuelMax, int pIdleFuelBurn, int pVision,
+        int pMovePower, MoveType pPropulsion, UnitActionType[] actions, WeaponModel[] weapons, double starValue)
+    {
+      super(pName, pRole, pChassis, cost, pAmmoMax, pFuelMax, pIdleFuelBurn, pVision, pMovePower, pPropulsion, actions, weapons, starValue);
+      type = pType;
+    }
+    public AWBWUnitModel(String pName, AWBWUnitEnum pType, UnitRoleEnum pRole, ChassisEnum pChassis, int cost, int pAmmoMax, int pFuelMax, int pIdleFuelBurn, int pVision,
+        int pMovePower, MoveType pPropulsion, ArrayList<UnitActionType> actions, ArrayList<WeaponModel> weapons, double starValue)
+    {
+      super(pName, pRole, pChassis, cost, pAmmoMax, pFuelMax, pIdleFuelBurn, pVision, pMovePower, pPropulsion, actions, weapons, starValue);
+      type = pType;
+    }
+
+    @Override
+    public UnitModel clone()
```

Comment:
Would it be possible to push `clone` up to `UnitModel` so it doesn't have to be re-implemented in each child?

Reply:
That sounds lovely, but the children need to be able to clone their extra values.
Do you have a way to solve that?

Reply:
Hrm. Could just have the parent `clone` handle the common attributes. The child `clone` can call the parent one and then finish off the extra things. I think that should work.